### ISSUE TITLE
Offer a capability to send prompts to an SQS or Pub/Sub or Kafka or maybe even our own PostHog for analytics

### ISSUE THREAD

**Issue by smpnet74** [continuedev/continue#646]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

We are looking for ways to continually improve our staff.  Understanding the nature of what is being asked of the model and training and upskilling our staff for frequently asked challenges is something we have been looking for in a product.  Given the ability, maybe via PostHog, or another queue technology like AWS SQS or GCP Pub/Sub to send the prompts in addition to the model would be enormously helpful.  In a perfect world, the product would work as is, but also give the ability to send the prompt to another destination that could be used for introspection and analytics.  The first choice would be AWS SQS but would accept any destination as long as we control it.

### Solution

If it was as simple as providing an API key and URL in config.json for a supported queue endpoint much like we have options for local and remote models.  AWS SQS would be first choice, but again, would accept any technology as long as it works!

**Comment by smpnet74** [comment]
While something built-in would be the first choice, I think this would be an acceptable solution as well.  Would you be able to help us thru the first example so we have something to build upon?

**Comment by sestinj** [comment]
Absolutely! I will add the hook later today and then share a quick example

**Comment by smpnet74** [comment]
The idea is not to make it overly complex and to be able to use the product as just another vscode extension.  We have thousands of developers so keeping it simple would be preference.  I am totally into allowing to send to a private posthog in addition to the posthog leveraged by continue.dev.  We can create all kinds of architecture about pulling the prompts from there that don't have to involve this codebase.

**Comment by sestinj** [comment]
@smpnet74 great! Glad you found a solution.

When you say to use as "just another VS Code extension", does this mean that ideally you don't have to setup a Continue server for your team, they can just download the extension entirely themselves? Asking because we're currently looking at removing this complexity, would be useful to know where you want to trade off control/flexibility with simplicity and ease of distribution.
